app/DAO/BakeryDatabase.scala

```diff
@@ -35,4 +35,100 @@ class BakeryDatabase @Inject() (
+        val t = ""
+
+        try {
+
+          /** the following must be done to add a new record/row to a table:
+            *    1. establish a connection to the database
+            *    2. connect to the table that will be updated (should this be a const val within this class?)
+            *    3. parse json object for column values to be inserted in the new row
+            *    4. insert the values to the table
+            *    4. commit the transaction, rollback if you can't
+            */
+          val statement =
+            conn.prepareStatement("INSERT INTO product() VALUES ()")
+          val insertResult = statement.executeQuery()
+        } catch {
+          case e: Exception => e.printStackTrace().toString
+        }
+
+        t
+      }
+    }
+  }
+
+  def getAllProducts: Future[String] = {
+    Future {
+      db.withConnection { conn =>
+        try {
+          val statement = conn.prepareStatement("SELECT * FROM product;")
+          val selectResult = statement.executeQuery()
+          var returnStr = ""
+          var idx = 1
+          while (selectResult.next()) {
+            returnStr += "Item " + idx + ":: id: " + selectResult.getString(
+              "id"
+            ) + ", name: " + selectResult.getString(
+              "name"
+            ) + ", quantity: " + selectResult.getInt(
+              "quantity"
+            ) + ", price: " + selectResult.getDouble(
+              "price"
+            ) + ", created at: " + selectResult.getTimestamp(
+              "created_at"
+            ) + ", updated at: " + selectResult.getTimestamp(
+              "updated_at"
+            ) + "\n"
+            idx += 1
+          }
+          returnStr
+        } catch {
+          case e: Exception => e.printStackTrace().toString
+        }
+      }
+    }
+  }
+
+  def getProductById(id: String): Future[Option[String]] = {
+    Future {
+      db.withConnection { conn =>
+        try {
+          val statement = {
+            conn.prepareStatement("SELECT * FROM product WHERE id::text = ?")
+          }
+          statement.setString(1, id)
+
+          println(statement.toString)
+          val selectResult = statement.executeQuery()
+          val result = if (selectResult.next()) {
+            Some(
+              "id: " + selectResult.getString(
+                "id"
+              ) + ", name: " + selectResult.getString(
+                "name"
+              ) + ", quantity: " + selectResult.getInt(
+                "quantity"
+              ) + ", price: " + selectResult.getDouble(
+                "price"
+              ) + ", created at: " + selectResult.getTimestamp(
+                "created_at"
+              ) + ", updated at: " + selectResult.getTimestamp(
+                "updated_at"
+              )
+            )
+          } else {
+            None
+          }
+          result
+        } catch {
+          case e: Exception => Option(e.printStackTrace().toString)
+        }
```

Comment:
Read about `Try()`, which is a Scala construct that you will see often. No action to take on this so far. Some links:

https://alvinalexander.com/scala/try-success-failure-example-reading-file/
https://medium.com/@shantanukamath/why-try-scala-error-handling-d71d2bd44d8c

The advantage of this over try-catch is that sometimes reading error handling is easier (with match case statements). Similar to what we did with options